packages/deploy-tooling/src/base/deploy.ts

```diff
@@ -281,6 +326,46 @@ async function getAbapService(config: AbapDeployConfig, logger: Logger): Promise
     return service;
 }
 
+/**
+ * Creates a new transport request using adt service.
+ *
+ * @param config - deployment configuration
+ * @param logger - reference to the logger instance
+ * @param provider - instance of the axios-extension abap service provider (it is created if not passed in)
+ * @returns transportRequest
+ */
+export async function createTransportRequest(
+    config: AbapDeployConfig,
+    logger: Logger,
+    provider?: AbapServiceProvider
+): Promise<string | void> {
+    if (!provider) {
+        provider = await getAbapServiceProvider(config, logger);
+    }
+
+    const adtService = await provider.getAdtService<TransportRequestService>(TransportRequestService);
+    if (adtService) {
+        const createTransportParams = {
```

Comment:
Can you add tsdoc @throws (https://tsdoc.org/pages/tags/throws/) to indicate deploy is not triggered if createTransportRequest failed with error.

Reply:
Updated